src/sync.rs

```diff
@@ -1,15 +1,33 @@
+#[cfg(all(not(feature = "std"), loom))]
+compile_error!("loom requires the standard library");
+
+//loom support
+
 #[cfg(loom)]
 pub(crate) mod atomic {
     pub(crate) use loom::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
 }
-
 #[cfg(loom)]
 pub(crate) use loom::thread::yield_now;
 
-#[cfg(not(loom))]
+//std imports
+
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) mod atomic {
     pub(crate) use std::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
 }
-
-#[cfg(not(loom))]
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) use std::thread::yield_now;
+
+//core imports
+
+#[cfg(all(not(loom), not(feature = "std")))]
```

Comment:
I don't think you need this to be separate. Just change the `use std::` above to `use core::`.